tests/phpunit/BitSetTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of BedrockProtocol.
+ * Copyright (C) 2014-2022 PocketMine Team <https://github.com/pmmp/BedrockProtocol>
+ *
+ * BedrockProtocol is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\network\mcpe\protocol;
+
+use PHPUnit\Framework\TestCase;
+use pocketmine\network\mcpe\protocol\serializer\BitSet;
+use pocketmine\network\mcpe\protocol\serializer\PacketSerializer;
+use function PHPUnit\Framework\assertTrue;
+
+class BitSetTest extends TestCase{
+
+	public function testBitSet() : void{
+		$writeTest = new BitSet(65);
+
+		$writeTest->set(0, true);
+		$writeTest->set(64, true);
+
+		$packetSerializer = PacketSerializer::encoder();
+		$writeTest->write($packetSerializer);
+
+		$packetSerializer = PacketSerializer::decoder($packetSerializer->getBuffer(), 0);
+		$readTest = BitSet::read($packetSerializer, 65);
+
+		assertTrue($this->setsEqual($writeTest, $readTest));
+	}
+
+	public function testBitSetConstructor() : void{
+		$test = new BitSet(65, [-9223372036854775808, 1]);
```

Comment:
```suggestion
		$test = new BitSet(65, [-9223372036854775807 - 1, 1]);
```
PHP is weird about this